src/ramanchada2/protocols/twinning.py

```diff
@@ -0,0 +1,283 @@
+            fit_res = spe.fit_peak_multimodel(
+                profile="Voigt", candidates=candidates, no_fit=no_fit
+            )
+            df = fit_res.to_dataframe_peaks()
+            df["sorted"] = abs(
+                df["center"] - self.reference_band_nm
+            )  # closest peak to 144
+            df_sorted = df.sort_values(by="sorted")
+
+            # we get actual y value, not height or amplitude
+            index_left = np.searchsorted(
+                spe.x, df_sorted["center"].iloc[0], side="left", sorter=None
+            )
+            index_right = np.searchsorted(
+                spe.x, df_sorted["center"].iloc[0], side="right", sorter=None
+            )
+            if index_right == index_left:
+                peak_intensity = spe.y[index_left]
+                peak_position = spe.x[index_left]
+            else:
+                peak_intensity = (spe.y[index_right] + spe.y[index_left]) / 2.0
+                peak_position = (spe.x[index_right] + spe.x[index_left]) / 2
+
+            # _label = "intensity = {:.3f} {} ={:.3f} amplitude={:.3f} center={:.1f}".format(
+            #    intensity_val,peak_intensity,df_sorted.iloc[0][peak_intensity],
+            #    df_sorted.iloc[0]["amplitude"],df_sorted.iloc[0]["center"])
+
+            return peak_intensity, peak_position, fit_res
+        except Exception as err:
+            print(err)
+            return None, None, None
+
+    def laser_power_regression(
+        self, df: SpectraFrame, boundaries=None, no_fit=False, source="spectrum"
+    ):
+        # fig, ax = plt.subplots(df.shape[0],1,figsize=(12,12))
+        r = 0
+        for index, row in df.iterrows():
+            spe = row[source]
+            if boundaries is None:
+                boundaries = (self.reference_band_nm - 50, self.reference_band_nm + 50)
+            peak_intensity, peak_position, fit_res = self.calc_peak_intensity(
+                spe, boundaries=boundaries, no_fit=no_fit
+            )
+            df.at[index, "peak_intensity"] = peak_intensity
+            df.at[index, "peak_position"] = peak_position
+            # spe.trim_axes(method='x-axis',boundaries=boundaries).plot(ax=ax[r], fmt=':',label=row["laser_power_mW"])
+            # if fit_res is not None:
+            #    fit_res.plot(ax=ax[r])
+            r = r + 1
+        # plt.savefig("test_twinning_peaks_{}.png".format(title))
+        # print(df[['laser_power_mW','peak_intensity']])
+        return LinearRegression().fit(
+            df[["laser_power_mW"]].values, df["peak_intensity"].values
+        )
+
+    def process(
+        self, spe: SpectraFrame, source="spectrum", target="spectrum_harmonized"
+    ):
+        spe.multiply(self.correction_factor, source=source, target=target)
+        spe.spe_area(
+            source=target, target="area_harmonized", boundaries=self.boundaries
+        )
+
+    def process_spectrum(self, spe: Spectrum):
+        return spe * self.correction_factor
```

Comment:
error: Unsupported operand types for * ("Spectrum" and "float")  [operator]

Reply:
remove process_spectrum